docs/core/tools/dotnet-nuget-trust.md

```diff
@@ -0,0 +1,319 @@
+  
+#### OPTIONS:
+
+- **`--allow-untrusted-root`**
+
+  Specifies if the certificate for the trusted signer should be allowed to chain to an untrusted root. This is not recommended.
+
+- **`-h|--help`**
+
+  Prints out a description of how to use the command.
+
+- **`-v, --verbosity <LEVEL>`**
+
+  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+
+- **`--configfile <PATH>`**
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+### `repository`
+
+Adds a trusted signer with the given name, based on the repository signature or countersignature of a signed package.
+
+#### Synopsis
+
+```dotnetcli
+dotnet nuget trust repository <name> <package> [OPTIONS]
+```
+
+#### Arguments
+
+- **`name`**
+
+  The name of existing trusted signer going be added. If `name` already exists in the configuration, the signature will be appended.
+
+- **`package`**
+
+  The given `package` should be local path to the signed .nupkg file.
+
+#### OPTIONS:
+
+- **`--allow-untrusted-root`**
+
+  Specifies if the certificate for the trusted signer should be allowed to chain to an untrusted root. This is not recommended.
+
+- **`-h|--help`**
+
+  Prints out a description of how to use the command.
+
+- **`-v, --verbosity <LEVEL>`**
+
+  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+
+- **`--configfile <PATH>`**
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+### `certificate`
+
+Adds a trusted signer with the given name, based on a certificate fingerprint.
+
+#### Synopsis
+
+```dotnetcli
+dotnet nuget trust certificate <name> <fingerprint> [OPTIONS]
+```
+
+#### Arguments
+
+- **`name`**
+
+  The name of existing trusted signer going be added. If a trusted signer with the given name already exists, the certificate item will be added to that signer. Otherwise a trusted author will be created with a certificate item from given certificate information.
+
+- **`<fingerprint>`**
+
+  The fingerprint of the certificate.
+
+#### OPTIONS:
+
+- **`--allow-untrusted-root`**
+
+  Specifies if the certificate for the trusted signer should be allowed to chain to an untrusted root. This is not recommended.
+
+- **`--algorithm <ALGORITHM>`**
+
+  Specifies the hash algorithm used to calculate the certificate fingerprint. Defaults to SHA256. Values supported are SHA256, SHA384 and SHA512.
+  
+- **`-h|--help`**
+
+  Prints out a description of how to use the command.
+
+- **`-v, --verbosity <LEVEL>`**
+
+  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+
+- **`--configfile <PATH>`**
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+### `source`
+
+Adds a trusted signer based on a given package source.
+
+#### Synopsis
+
+```dotnetcli
+dotnet nuget trust source <name> [source-url] [--owners <List>] [OPTIONS]
```

Comment:
IF source-url is an option rather than an argument, it should be prefixed by `--`.